src/flow-hash.c

```diff
@@ -692,6 +692,7 @@ static inline void MoveToWorkQueue(ThreadVars *tv, FlowLookupStruct *fls,
         f->fb = NULL;
         f->next = NULL;
         FlowQueuePrivateAppendFlow(&fls->work_queue, f);
+        FLOWLOCK_UNLOCK(f);
```

Comment:
This looks ok but got me confused at first. FlowQueuePrivateAppendFlow is used without unlocking on some other part of the code. Only work_queue has the lock at start because of injection from external threads. Should we use a dedicated function for work_queue to avoid confusion ?

Reply:
`FlowQueue` API uses its own lock logic independent of the `Flow` objects with passes around.